enhancements/machine-api/aws-efa-networking.md

```diff
@@ -34,14 +37,21 @@ OpenShift. To enable this, we must provide support for the feature within the Ma
 
 ### Goals
 
-- Enable worker Machines to attach EFA network interfaces as their primary network interface
-- Enable autoscaling of Machines that leverage high performance network interfaces
+- Enable Non Control Plane Machines to attach EFA network interfaces as their primary network interface
+- Enable autoscaling of Machines that use EFA network interfaces
+- Ensure that RHCOS contains the required kernel module (efa) to support the new network interface type
+- Enable a mixture of MachineSet types for both EFA and non-EFA enabled instances
 
 ### Non-Goals
 
-- Support and or configuration of the prerequisite requirements for using EFA
+- Support and/or configuration of the prerequisite requirements for using EFA
   - Configuration of security groups to allow the inter-node traffic
   - Deployment of any Libfabric/MPI software prerequisites
+  - Configuration of Huge Pages
+- Allowing day 1 worker Machine configuration for EFA
+- Allowing Control Plane Machines to be configured for EFA
+- Support and/or configuration of other HPC related features in AWS
+  - Eg. Placing Machines into placement groups
```

Comment:
is this required to actually successfully use the EFA feature?

Reply:
@deads2k 

> is this required to actually successfully use the EFA feature?

Technically, [no](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-start.html#efa-start-instances):

> **Note**
> 
> It is not an absolute requirement to launch your EFA-enabled instances into a cluster placement group. However, we do recommend running your EFA-enabled instances in a cluster placement group as it launches the instances into a low-latency group in a single Availability Zone.

The question is whether an EFAs-without-placement-groups feature will satisfy the customer.

Reply:
Just to fill in some offline discussions. We will need to deliver placement groups in order to meet the latency needs of the customer who has driven this implementation. As such we will hold this feature from landing until we can delivery both at the same time, however placement groups will come via a separate enhancement given the size of that work and the significant design considerations which need to be accounted for. We should probably call out that placement groups are a noted concern but outside of scope for this enhancement.